### ISSUE TITLE
[FEATURE] Add special k repo releases as a download option

### ISSUE THREAD

**Issue by zany130** [sonic2kk/steamtinkerlaunch#505]
https://github.com/SpecialKO/SpecialK/releases

EDIT: tried to add this myself by changing the url in https://github.com/frostworx/steamtinkerlaunch/blob/96b3b75fb220e514964659246e797e83c6be4b16/steamtinkerlaunch#L2349=

to https://github.com/SpecialKO/SpecialK/releases/latest/download/SpecialK.7z

but that didn't work for some reason.

tested to see if wget could get that link and it does

**Comment by frostworx** [comment]
nice, thanks for the info. will try to add it soon

**Comment by zany130** [comment]
seems like the stable version here https://sk-data.special-k.info/SpecialK.7z also was updated and now works on proton so that could probably be the new default with latest github as an option?

**Comment by frostworx** [comment]
yeah, this definitely makes sense.
Already initially started with an implementation for the github repo, but now will prioritize [this merge request](https://github.com/frostworx/steamtinkerlaunch/pull/509) to avoid conflict.
The idea is to have the option between "default", "latest" _(and a list of all versions available on github autocreated)_
with the default "default" pointing to the version on the project page, and "latest" being the latest on github.

**Comment by sonic2kk** [comment]
If I'm not mistaken, SpecialK can be used for NieR:Automata right? If so, and if that SpecialK plugin works with Proton, I'm happy to test this as well once it's implemented.

Also, if it does work for NieR:Automata, this will be a really exciting update (at least for me)! 😄

**Comment by frostworx** [comment]
SpecialK ist already implemented since some time (see [wiki](https://github.com/frostworx/steamtinkerlaunch/wiki/SpecialK))
It only didn't have a reliable download source until now.
Haven't touched SpecialK for a while, but I guess proton compatibility has even improved since.

**Comment by frostworx** [comment]
steamtinkerlaunch already makes use of this https://www.pcgamingwiki.com/wiki/List_of_games_compatible_with_Special_K#Compatibility_list
to get the render api. Your Nier: Automata is listed :)

**Comment by frostworx** [comment]
the download rewrite part is already more or less complete _(still offline)_
I'm not sure yet if parts from the `prepareDiscordSpecialK` function are still required, so it needs some more testing
_(f.e. it does not seem to be necessary anymore to enable `UsingWINE` in the specialk ini - at least I think so :))_

some more possible features to add while working on it anyway:
- adding an "autoupdate" function per game would probably make sense, so the used specialk version for a game will be always overwritten _(only makes sense when selected SPEKVERS is either "default" or latest)_
 - maybe some specialk features could be used to show as additional information in steamtinkerlaunch _(f.e. its steamapi - i.e. showing achievements)_
- ~~probably makes sense to work with symlinks instead of copying the specialk dlls into the game dir~~ it doesn't. debugging files would have to be checked again or they would not match, also the target dll names might also come from somewhere else
- more?

**Comment by frostworx** [comment]
[main commit just landed](https://github.com/frostworx/steamtinkerlaunch/commit/18d2d15dce4e49c1768a70673c75a5fe5a65be01).
happy testing!

also bumped the [SpecialK](https://github.com/frostworx/steamtinkerlaunch/wiki/SpecialK) wiki a bit

**Comment by sonic2kk** [comment]
Woohoo! Should hopefully have a chance to test tonight or tomorrow :partying_face:

**Comment by zany130** [comment]
> (f.e. it does not seem to be necessary anymore to enable `UsingWINE` in the specialk ini - at least I think so :))

from my testing some games still need it but not all. I think it should still be done and the user can try disabling it if anything. and yeah nier automata had a special version of special k made for it @sonic2kk i don't know if its still recommended to use that version or the latest specialk though. it was called FAR i think. (I think i used the far mod when i played nier automata on linux a few years ago, back when it was one of the first proton whitelisted games great game)

**Comment by sonic2kk** [comment]
I think it's the latest SpecialK now going by this https://wiki.special-k.info/SpecialK/Custom/FAR but I could be misunderstanding. Still nice to know though thank you :smile: If you know whether SpecialK or FAR should be used let me know, I thought there was an "official" SpecialK Steam Guide for NieR:Automata but I can't find it now

EDIT: Something else I found on the SpecialK docs, the part about the HD texture mod folder being renamed to `SK` instead of `FAR` was another tip off that SpecialK should be used in place of FAR, but I am no expert

**Comment by sonic2kk** [comment]
So I updated STL and launched NieR:Automata to test, but I'm only seeing `default` listed as a version (there is a blank entry underneath it but it isn't labelled). From looking at the wiki and code it seems like there should be two: `default` and `latest`. Even after removing `/dev/shm/steamtinkerlaunch` there was no change. I checked a couple of other games and they display the same, just `default`, not `default` and `latest`

![image](https://user-images.githubusercontent.com/7917345/176779863-7642c4a4-0d9a-4829-8d86-842ec42c20ef.png)

I'll attach my log though I'm not sure if there's anything in it that's useful (I didn't try installing Special K yet, I think `latest` is the version I would want to use?). This could also just be my own misunderstandings :sweat_smile: 

[524220.log](https://github.com/frostworx/steamtinkerlaunch/files/9023480/524220.log)

**Comment by zany130** [comment]
I would use either the latest stable (default maybe it should be named stable?) or git release. It seems the git releases are missing from the drop-down

**Comment by frostworx** [comment]
hi,
sorry for short replies...

> UsingWINE

The commit above already covers that - it is still enabled automatically _(like before)_

> just default, not default and latest

Only tested it quickly. should be easy to fix, thanks for reporting

> I prefer default and latest, because steamtinkerlaunch simply can't decide what stable is (especially as packages still are unversioned)

> It seems the git releases are missing from the drop-down

https://github.com/frostworx/steamtinkerlaunch/wiki/SpecialK#versions=

**Comment by frostworx** [comment]
> just default, not default and latest

should be fixed by latest commit. I forgot to remove some debug lines previously.

**Comment by zany130** [comment]
still only shows default 
![Screenshot_10](https://user-images.githubusercontent.com/25832539/176948759-32ada028-8be2-45b2-ba2f-976f3392a4aa.png)

EDIT: really strange but I was able to get it to show if I run `steamtinkerlaunch settings` in a terminal instead of letting it load from steam. If I then select latest and save it it will keep that option when I load it through steam but if I go back to default latest disappears again and I have to run `steamtinkerlaunch settings` in a terminal

so it seems the list is having issues getting populated when it is ran through steam but it works fine through the terminal

EDIT2: running steamtinkerlaunch as a compat tool in steam BTW

**Comment by frostworx** [comment]
heh, I really shouldn't make quick changes when I don't have the time for it.
the bug is simple, I simply added the new function createDLSpekList to the wrong function `startSettings`.
it belongs to the top of function `MainMenu`.
didn't want to code anything at the weekend, but will push a fix in a few minutes.
thx for re-testing.

**Comment by frostworx** [comment]
boing. have a nice weekend :)

**Comment by zany130** [comment]
yup works as expected and it shows all the github releases too (assuming you enable `USESPECIALK`  first) going to go ahead on close this since everything works as expected

**Comment by sonic2kk** [comment]
Just want to confirm it works for me as well :smile:  

![image](https://user-images.githubusercontent.com/7917345/176957109-360b0d46-67df-40a9-bbd7-80438748db0a.png)

Very awesome, thank you :D

**Comment by sonic2kk** [comment]
Not sure if it's STL specific but NieR:Automata seems to crash when trying out Special K. I'll have to try out some other games and see how it goes.

EDIT: Seems to be working with NieR Replicant but I couldn't get it to respond to input and it crashed with this dialog

![image](https://user-images.githubusercontent.com/7917345/177012046-a2549783-479f-4495-ab7a-fc4ede20e2f3.png)

Likely not anything STL specific though :-) And I'll put Automata down to some weirdness. I'll play around with Automata sometime in the future and see if I can get SpecialK to work but I did see someone report the exact same problem with SpecialK on Windows with Automata but I can't find it now. Automata might be better with FAR like zany130 pointed out.

EDIT2: So I remembered I did some system updates and hadn't restarted yet, when I restarted my PC and tried Replicant again, it worked almost completely without issue (alt+tab from fullscreen breaks the game, but Gamescope/playing borderless fixed it. Probably not even anything Wine specific).

Seems like SpecialK works and the problems I had with Automata are specific to that game and not STL :-)
